cpp/cmake/thirdparty/get_libcugraphops.cmake

```diff
@@ -14,26 +14,50 @@
 # limitations under the License.
 #=============================================================================
 
-function(find_and_configure_cugraphops)
+set(CUGRAPH_MIN_VERSION_cugraph_ops "${CUGRAPH_VERSION_MAJOR}.${CUGRAPH_VERSION_MINOR}.00")
+set(CUGRAPH_BRANCH_VERSION_cugraph_ops "${CUGRAPH_VERSION_MAJOR}.${CUGRAPH_VERSION_MINOR}")
 
-    if(TARGET cugraphops::cugraphops)
-        return()
-    endif()
+function(find_and_configure_cugraph_ops)
 
-    rapids_find_generate_module(cugraphops
-        HEADER_NAMES            graph/sampling.hpp
-        LIBRARY_NAMES           cugraph-ops++
-        INCLUDE_SUFFIXES        cugraph-ops
-        BUILD_EXPORT_SET    cugraph-exports
-        INSTALL_EXPORT_SET  cugraph-exports
-    )
+    set(oneValueArgs VERSION FORK PINNED_TAG CLONE_ON_PIN)
+    cmake_parse_arguments(PKG "" "${oneValueArgs}" "" ${ARGN} )
+
+    if(PKG_CLONE_ON_PIN AND NOT PKG_PINNED_TAG STREQUAL "branch-${CUGRAPH_BRANCH_VERSION_cugraph_ops}")
+        message("Pinned tag found: ${PKG_PINNED_TAG}. Cloning cugraph-ops locally.")
+        set(CPM_DOWNLOAD_cugraph-ops ON)
+    endif()
 
-    rapids_find_package(cugraphops
-        REQUIRED
-        BUILD_EXPORT_SET    cugraph-exports
-        INSTALL_EXPORT_SET  cugraph-exports
+    rapids_cpm_find(cugraph-ops ${PKG_VERSION}
```

Comment:
What we might want to is provide more control over building cugraph-ops from source which would allow us to provide error messages and instructions as needed.

```cmake
option(CUGRAPH_BUILD_CUGRAPHOPS_FROM_SOURCE OFF)

if(NOT CUGRAPH_BUILD_CUGRAPHOPS_FROM_SOURCE)
  rapids_find_generate_module(cugraph-ops)
  rapids_find_package(cugraph-ops)
  if(NOT TARGET cugraph-ops::cugraph-ops++)
    message(FATAL_ERROR "The current enviornment you are building in doesn't have cugraph-ops. If using conda please read <X.Y> to properly configure the enviornment. If you are building from source and have access to the cugraph-ops repositry please enable `CUGRAPH_BUILD_CUGRAPHOPS_FROM_SOURCE`)
  endif()
endif()

if(CUGRAPH_BUILD_CUGRAPHOPS_FROM_SOURCE AND
   NOT TARGET cugraph-ops::cugraph-ops++)
    rapids_cpm_find(cugraph-ops ${PKG_VERSION} .... )
endif()
```

Reply:
As I mentioned in slack, I worry this would be introducing code that is effectively never tested -- neither CI nor devs who primarily use conda will set `-D CUGRAPH_BUILD_CUGRAPHOPS_FROM_SOURCE=ON`